app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/adapter/SwapTokenAItemDecoration.kt

```diff
@@ -0,0 +1,99 @@
+package org.p2p.wallet.swap.ui.jupiter.tokens.adapter
+
+import androidx.annotation.Px
+import androidx.core.view.forEach
+import androidx.recyclerview.widget.RecyclerView
+import android.graphics.Canvas
+import android.view.View
+import com.google.android.material.shape.ShapeAppearanceModel
+import org.p2p.uikit.components.finance_block.FinanceBlockViewHolder
+import org.p2p.uikit.utils.drawable.shape.rippleForeground
+import org.p2p.uikit.utils.drawable.shape.shapeBottomRounded
+import org.p2p.uikit.utils.drawable.shape.shapeOutline
+import org.p2p.uikit.utils.drawable.shape.shapeRectangle
+import org.p2p.uikit.utils.drawable.shape.shapeRoundedAll
+import org.p2p.uikit.utils.drawable.shape.shapeTopRounded
+import org.p2p.uikit.utils.toPx
+import org.p2p.wallet.swap.jupiter.domain.model.SwapTokenModel
+
+class SwapTokenAItemDecoration(
```

Comment:
why do we need this ? we already have decorators with rounded items ? why not to reuse them ?

Reply:
old decorator for old adapters and old items and have a lot of interfaces